lib/facter/facts/solaris/hypervisors/zone.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Facts
+  module Solaris
+    module Hypervisors
+      class Zone
+        FACT_NAME = 'hypervisors.zone'
+
+        def initialize
+          @log = Facter::Log.new(self)
+        end
+
+        def call_the_resolver
+          fact_value = current_zone
+
+          Facter::ResolvedFact.new(FACT_NAME, fact_value)
+        end
+
+        def current_zone
+          current_zone_name = Facter::Resolvers::Solaris::ZoneName.resolve(:current_zone_name)
+          return unless current_zone_name
+
+          zones = Facter::Resolvers::Solaris::Zone.resolve(:zone)
+          return Facter::ResolvedFact.new(FACT_NAME, nil) unless zones
```

Comment:
here should be **return nil** otherwise the value for this fact will be another ResolvedFact object

Reply:
Updated.